File: src/main/java/lambda/orders/services/CustomerService.java
package lambda.orders.services;

import lambda.orders.models.Customer;

import java.util.List;

public interface CustomerService
{
	List<Customer> findAll();

	Customer findCustomerById(long id);

	Customer findCustomerByName(String name);

	void delete(long id);

	Customer save(Customer customer);

	Customer update(Customer customer, long id);
}

File: src/main/kotlin/com/wkdrabbit/crudycustomers/service/CustomerService.kt
package com.wkdrabbit.crudycustomers.service

import com.wkdrabbit.crudycustomers.model.Customers
import java.util.ArrayList


interface CustomerService {

    fun findAll(): ArrayList<Customers>

    fun findCustomersById(id: Long): Customers

    fun findCustomersByName(name: String): Customers

    fun delete(id: Long)

    fun save(restaurant: Customers): Customers

    fun update(restaurant: Customers, id: Long): Customers
}